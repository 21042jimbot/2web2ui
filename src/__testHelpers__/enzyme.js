export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
