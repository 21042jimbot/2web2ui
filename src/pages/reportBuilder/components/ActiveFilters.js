import React from 'react';
import _ from 'lodash';
import { Box, Inline, Stack, Tag, Text } from 'src/components/matchbox';
import { Comparison, Emphasized } from 'src/components/text';
import { getIterableFormattedGroupings, getGroupingFields } from '../helpers';
import styles from './ReportOptions.module.scss';

export function ActiveFilters({ filters, handleFilterRemove }) {
  const filtersWithIndex = filters.map((value, index) => ({ ...value, index }));
  const groupedFilters = _.groupBy(filtersWithIndex, 'type');

  return (
    <Stack>
      {Object.keys(groupedFilters).map(key => (
        <Inline key={`filter_group_${key}`}>
          <div>{key}</div>
          <div>
            <strong className={styles.Conditional}>equals</strong>
            {groupedFilters[key].map(({ value, index }) => (
              <Tag
                className={styles.TagWrapper}
                key={`tag_${index}`}
                onRemove={handleFilterRemove ? () => handleFilterRemove(index) : undefined}
              >
                {value}
              </Tag>
            ))}
          </div>
        </Inline>
      ))}
    </Stack>
  );
}

export function ActiveFiltersV2({ filters, handleFilterRemove }) {
  const groupings = getGroupingFields(getIterableFormattedGroupings(filters));

  // TODO: Once `Inline` supports the `data-id` prop, extra <div> elements with `data-id` will no longer needed.
  // Matchbox issue: https://github.com/SparkPost/matchbox/issues/661
  return (
    <div data-id="active-filter-tags">
      <Inline space="200">
        {groupings.map((grouping, groupingIndex) => {
          return (
            <div data-id="active-filter-group">
              <Inline
                key={`grouping-${groupingIndex}`}
                as="span"
                display="inline-flex"
                paddingY="100"
                paddingX="200"
                backgroundColor="gray.100"
                space="0"
              >
                {grouping.filters.map((filter, filterIndex) => {
                  return (
                    <Box
                      key={`filter-${groupingIndex}-${filterIndex}`}
                      paddingY="100"
                      paddingX="200"
                      backgroundColor="gray.100"
                      data-id="active-filter"
                    >
                      <Inline as="span" space="200">
                        <Text fontSize="200" as="span">
                          {filter.label}
                        </Text>

                        <Text fontWeight="500" as="span" fontSize="200">
                          <Emphasized>{getCompareByText(filter.compareBy)}</Emphasized>
                        </Text>

                        {filter.values.map((rawValue, valueIndex) => {
                          const value = typeof rawValue === 'object' ? rawValue.value : rawValue;
                          const onRemoveFn = handleFilterRemove
                            ? () => handleFilterRemove({ groupingIndex, filterIndex, valueIndex })
                            : undefined;

                          return (
                            <Tag
                              key={`tag-${groupingIndex}-${filterIndex}-${valueIndex}`}
                              onRemove={onRemoveFn}
                            >
                              {value}
                            </Tag>
                          );
                        })}

                        {filter.hasComparisonBetweenFilters ? (
                          <Box marginX="200">
                            <Comparison>{grouping.type}</Comparison>
                          </Box>
                        ) : null}
                      </Inline>
                    </Box>
                  );
                })}

                {grouping.hasAndBetweenGroups ? (
                  <Box marginX="200">
                    <Comparison>And</Comparison>
                  </Box>
                ) : null}
              </Inline>
            </div>
          );
        })}
      </Inline>
    </div>
  );
}

function getCompareByText(compareByVal) {
  switch (compareByVal) {
    case 'eq':
      return 'is equal to';
    case 'notEq':
      return 'is not equal to';
    case 'like':
      return 'contains';
    case 'notLike':
      return 'does not contain';
    default:
      throw new Error(`${compareByVal} is not a valid comparison value.`);
  }
}
