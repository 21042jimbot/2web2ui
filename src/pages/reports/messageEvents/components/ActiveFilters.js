/* eslint-disable */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Panel, Tag } from '@sparkpost/matchbox';
import { getMessageEvents, removeFilter } from 'src/actions/messageEvents';
import { snakeToFriendly } from 'src/helpers/string';
import _ from 'lodash';

export class ActiveFilters extends Component {
  static defaultProps = {
    search: {}
  };

  renderTags = ({ key, label, itemToString }) => {
    const { search } = this.props;

    if (!search[key]) {
      return null;
    }

    return search[key].map((item) => (
      <Tag onRemove={() => this.handleRemove({ key, item })} key={item}>
        {label}: {itemToString ? itemToString(item) : item}
      </Tag>
    ));
  }

  handleRemove = (filter) => {
    this.props.removeFilter(filter);
  }

  render() {
    const { recipients, events } = this.props.search;

    if (!recipients.length && !events.length) {
      return null;
    }

    return (
      <Panel.Section>
        <small>Filters: </small>
        {this.renderTags({ key: 'recipients', label: 'Recipient' })}
        {this.renderTags({ key: 'events', label: 'Event', itemToString: snakeToFriendly })}
      </Panel.Section>
    );
  }
}

const mapStateToProps = (state, props) => ({
  search: state.messageEvents.search
});

export default connect(mapStateToProps, { removeFilter, getMessageEvents })(ActiveFilters);
