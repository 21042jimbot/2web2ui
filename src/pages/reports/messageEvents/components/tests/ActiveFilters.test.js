import React from 'react';
import { shallow } from 'enzyme';
import { ActiveFilters } from '../ActiveFilters';

describe('Component: ActiveFilters', () => {
  let wrapper;
  const props = {
    removeFilter: jest.fn(),
    updateMessageEventsSearchOptions: jest.fn(),
    search: {
      dateOptions: {
        to: 'to-date',
        from: 'from-date'
      },
      message_ids: []
    }
  };

  beforeEach(() => {
    wrapper = shallow(<ActiveFilters {...props} />);
  });

  it('should render nothing with no filters', () => {
    expect(wrapper.html()).toBe(null);
  });

  it('should render filters correctly', () => {
    const events = ['bounce', 'click', 'spam_complaint'];
    const message_ids = ['101', 102];
    const campaigns = [];
    wrapper.setProps({ search: { events, message_ids, campaigns }});
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle remove', () => {
    const events = ['bounce', 'click', 'spam_complaint'];
    wrapper.setProps({ search: { events }});
    wrapper.find('Tag').at(1).simulate('remove');
    expect(props.removeFilter).toHaveBeenCalledWith({ item: 'click', key: 'events' });
  });

  it('should handle remove all', () => {
    const events = ['bounce', 'click', 'spam_complaint'];
    wrapper.setProps({ search: { ...props.search, events }});
    wrapper.instance().handleRemoveAll();
    expect(props.updateMessageEventsSearchOptions).toHaveBeenCalledWith({
      ...props.search,
      events: [],
      message_ids: []
    });
  });
});
