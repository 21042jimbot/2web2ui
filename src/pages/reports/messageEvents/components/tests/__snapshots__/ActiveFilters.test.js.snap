// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ActiveFilters should render filters correctly 1`] = `
<Panel.Section
  actions={
    Array [
      Object {
        "color": "blue",
        "content": "Clear All Filters",
        "onClick": [Function],
      },
    ]
  }
>
  <small>
    Filters: 
  </small>
  <Tag
    key="0"
    onRemove={[Function]}
  >
    Event
    : 
    Bounce
  </Tag>
  <Tag
    key="1"
    onRemove={[Function]}
  >
    Event
    : 
    Click
  </Tag>
  <Tag
    key="2"
    onRemove={[Function]}
  >
    Event
    : 
    Spam Complaint
  </Tag>
  <Tag
    key="0"
    onRemove={[Function]}
  >
    Message
    : 
    101
  </Tag>
  <Tag
    key="1"
    onRemove={[Function]}
  >
    Message
    : 
    102
  </Tag>
</Panel.Section>
`;
